packages/ec-linear/src/recombinator/errors.rs

```diff
@@ -0,0 +1,95 @@
+use std::{
+    error::Error,
+    fmt::{Debug, Display},
+};
+
+use miette::{Diagnostic, LabeledSpan, Severity, SourceCode};
+
+#[derive(Debug, thiserror::Error, Diagnostic)]
+#[error("Attempted to perform TwoPointXo on genomes of different lengths {0} and {1}")]
+#[diagnostic(help = "Ensure your genomes are of uniform length")]
+pub struct DifferentGenomeLength(pub usize, pub usize);
+
+#[derive(Debug)]
+pub enum CrossoverGeneError<E> {
+    DifferentGenomeLength(DifferentGenomeLength),
+    Crossover(E),
+}
```

Comment:
I feel like some sort of comment is warranted here explaining what the `Crossover` variant is doing. This is especially true since the rest of this file (nearly 100 lines) is really about dealing with the fact that we couldn't just use things like `transparent` here, but had implement things like `source` ourselves.